astropy/wcs/src/astropy_wcs.c

```diff
@@ -799,7 +780,7 @@ static PyTypeObject WcsType = {
 int _setup_wcs_type(
     PyObject* m) {
 
-  if (PyType_Ready(&WcsType) < 0)
+  if (PyType_Ready((PyTypeObject *)&WcsType) < 0)
     return -1;
```

Comment:
Github isn't letting me put a suggestion here for some reason. But:

get rid of `PyType_Ready` - this is included in `PyType_FromSpec`

Add

```
WcsType = PyType_FromSpec(&WCSBaseClass_spec);
if (WcsType == NULL) return -1;
```

Essentially `PyType_FromSpec` just needs to go in a function (usually the one that gets called setting up the module) rather than at global C scope (which is only for declarations)